src/USER-CONP/compute_conp_vector.cpp

```diff
@@ -303,30 +329,41 @@ void ComputeConpVector::create_taglist() {
     idispls[i] = idispls[i - 1] + igroupnum_list[i - 1];
   }
 
-  taglist = std::vector<int>(ngroup);
+  std::vector<int> taglist = std::vector<int>(ngroup);
   MPI_Allgatherv(&taglist_local.front(), igroupnum_local, MPI_LMP_TAGINT,
                  &taglist.front(), &igroupnum_list.front(), &idispls.front(),
                  MPI_LMP_TAGINT, world);
   // must be sorted for compatibility with fix_charge_update
   std::sort(taglist.begin(), taglist.end());
+
+  int const tag_max = taglist[ngroup - 1];
+  tag_to_iele = std::vector<int>(tag_max + 1, -1);
+  for (size_t i = 0; i < taglist.size(); i++) {
+    tag_to_iele[taglist[i]] = i;
+  }
 }
 
 /* ---------------------------------------------------------------------- */
 
 void ComputeConpVector::update_mpos() {
-  int const nmax = atom->nmax;
+  MPI_Barrier(world);
```

Comment:
Was machen die MPI_Barrier hier genau?

Reply:
Das wird für die Timings in den Logs gebraucht. Können wir später rausnehmen.